x-pack/plugins/uptime/server/rest_api/index_state/get_index_status.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { UMServerLibs } from '../../lib/lib';
+import { UMRestApiRouteFactory } from '../types';
+import { REST_API_URLS } from '../../../../../legacy/plugins/uptime/common/constants/rest_api';
+
+export const createGetIndexStatusRoute: UMRestApiRouteFactory = (libs: UMServerLibs) => ({
+  method: 'GET',
+  path: REST_API_URLS.INDEX_STATUS,
+  validate: false,
+  options: {
+    tags: ['access:uptime'],
+  },
+  handler: async ({ callES }, _context, _request, response): Promise<any> => {
+    try {
+      return response.ok({
+        body: {
+          ...(await libs.requests.getIndexStatus({ callES })),
+        },
+      });
+    } catch (e) {
+      return response.internalError({ body: { message: e.message } });
```

Comment:
@andrewvc has expressed reservations about returning descriptive error messages, this might be something we want to discuss here.

Reply:
i am not sure what do you mean by that?

Reply:
@andrewvc would you have any issue with returning this message? I thought I remembered you expressing concern about security implications of exposing error messages in API responses.

Reply:
I created https://github.com/elastic/uptime/issues/156 to address this.